components/post_view/channel_intro_message/channel_intro_message.tsx

```diff
@@ -588,3 +666,81 @@ function createBoardsButton(channel: Channel, boardComponent?: PluginComponent)
         </button>
     );
 }
+
+function createFavoriteButton(isFavorite: boolean, toggleFavorite: () => void, classes?: string) {
+    let favoriteText;
+    if (isFavorite) {
+        favoriteText = (
+            <FormattedMessage
+                id='channel_info_rhs.top_buttons.favorited'
+                defaultMessage='Favorited'
+            />);
+    } else {
+        favoriteText = (
+            <FormattedMessage
+                id='channel_info_rhs.top_buttons.favorite'
+                defaultMessage='Favorite'
+            />);
+    }
+    return (
+        <button
+            id='toggleFavoriteIntroButton'
+            className={`intro-links ${isFavorite ? 'active' : ''}  ${classes}`}
+            onClick={toggleFavorite}
+            aria-label={'Favorite'}
+        >
+            {isFavorite ? <StarIcon size={24}/> : <StarOutlineIcon size={24}/>}
+            {favoriteText}
+        </button>
+    );
+}
+
+function createNotificationButton(channel: Channel, user: UserProfileRedux, isNotificationsOpen: boolean, classes?: string) {
```

Comment:
this is not needed ... the implementing component can just use the `ToggleModalButton` directly